src/openMVG/sfm/pipelines/sequential/sequential_SfM.cpp

```diff
@@ -401,17 +405,10 @@ bool SequentialSfMReconstructionEngine::Process()
       char values[20];
       std::ifstream ifs;
 
-      if(!std::system("nproc >values.txt"))
-      {
-        ifs.open("values.txt");
-        if(ifs.is_open())
-        {
-          ifs.get(values, 20, '\n');
-          ifs.close();
-          _tree.put("hardware.cpu.cores", values);
-        }
-      }
+      // CPU cores
+      _tree.put("hardware.cpu.cores", sysconf(_SC_NPROCESSORS_ONLN));
 
+      // CPU frequency
```

Comment:
It can be a nice idea. But `#include <cpufreq.h>` does not work here. I tried `locate cpufreq.h` and ended up with bunch of `/usr/src/linux-headers-*/include/linux/cpufreq.h`. So I tried back with `#include <linux/cpufreq>` and nothing found again.